orion/schema.graphql

```diff
@@ -0,0 +1,18 @@
+# -----------------------------------------------
+# !!! THIS FILE WAS GENERATED BY TYPE-GRAPHQL !!!
+# !!!   DO NOT MODIFY THIS FILE BY YOURSELF   !!!
+# -----------------------------------------------
+
+type Mutation {
+  addVideoView(videoID: ID!): VideoInfo!
+}
+
+type Query {
+  batchedVideoViews(videoIDList: [ID!]!): [VideoInfo!]!
```

Comment:
I think this array might be sparse, for example if there are some IDs which don't exist in the list, also as it is right now it is not consistent with `VideoViews` which can be `null`, if there is a good reason for `batchedVideoViews` to be defined like it is then I think we should change `videoViews` to be consistent with it

Reply:
Yeah, I spent some time thinking about how this should look like. My initial approach was the one you're proposing I think: return `[VideoInfo]!`.

However, I figured that I'm not sure if this makes sense. The issue is that returning `null` doesn't really say anything. I think the client shouldn't rely just on the order of the returned array. Parsing the result will actually require doing a search - go through the results and match the video IDs we're looking for with the returned `VideoInfo`. In that case, returning additional `null` values from the resolver doesn't really bring any benefit or additional info.

For the `videoViews` I think it actually makes sense - with that endpoint you're asking for a specific ID so you either get a result or `null`. I guess I see those two endpoints as having a different purpose - `videoViews` is asking for something specific, while the `batchedVideoViews` is more like a search.

Let me know if that makes sense or whether you're like to discuss this on a call. Based on the result of our discussion I'll add some comments to those resolvers to better explain the rationale behind them.